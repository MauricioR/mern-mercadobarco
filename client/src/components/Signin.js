import React, { Component } from 'react';
import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

class Signin extends Component {
  render() {
    return (
      <Form>
        <FormGroup>
          <Label for="username">Username</Label>
          <Input type="text" name="name" id="username" />
        </FormGroup>
        <FormGroup>
          <Label for="password">Password</Label>
          <Input type="password" name="password" id="password" />
        </FormGroup>
        <Button>Sign in</Button>
      </Form>
    );
  }
}

export default Signin;
