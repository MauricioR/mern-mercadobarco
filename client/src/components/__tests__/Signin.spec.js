import React from 'react';
import { shallow, mount } from 'enzyme';

import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

import Signin from '../Signin';

describe('Testing Signin Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Signin />);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render username label', () => {
    expect(wrapper.find(Label).filterWhere((item) => {
      return item.prop('for') === 'username';
    })).toHaveLength(1);
  });

  it('should render username input', () => {
    expect(wrapper.find(Input).filterWhere((item) => {
      return item.prop('id') == 'username';
    })).toHaveLength(1);
  });

  it('should render password label', () => {
    expect(wrapper.find(Label).filterWhere((item) => {
      return item.prop('for') === 'password';
    })).toHaveLength(1);
  });

  it('should render password input', () => {
    expect(wrapper.find(Input).filterWhere((item) => {
      return item.prop('id') == 'password';
    })).toHaveLength(1);
  });

});
