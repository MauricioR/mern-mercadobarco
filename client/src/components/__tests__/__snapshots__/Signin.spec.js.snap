// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Signin Component should render correctly 1`] = `
<Signin>
  <Form
    tag="form"
  >
    <form
      className=""
    >
      <FormGroup
        tag="div"
      >
        <div
          className="position-relative form-group"
        >
          <Label
            for="username"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              className=""
              htmlFor="username"
            >
              Username
            </label>
          </Label>
          <Input
            id="username"
            name="name"
            type="text"
          >
            <input
              className="form-control"
              id="username"
              name="name"
              type="text"
            />
          </Input>
        </div>
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <div
          className="position-relative form-group"
        >
          <Label
            for="password"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              className=""
              htmlFor="password"
            >
              Password
            </label>
          </Label>
          <Input
            id="password"
            name="password"
            type="password"
          >
            <input
              className="form-control"
              id="password"
              name="password"
              type="password"
            />
          </Input>
        </div>
      </FormGroup>
      <Button
        color="secondary"
        tag="button"
      >
        <button
          className="btn btn-secondary"
          onClick={[Function]}
        >
          Sign in
        </button>
      </Button>
    </form>
  </Form>
</Signin>
`;
